### ISSUE TITLE
[node-js] Rows to a data frame. Unreliable arbitrary assignment of data to columns

### ISSUE THREAD

**Issue by volodymyrprokopyuk** [pola-rs/polars#3823]
## Context
- Manjaro Linux OS
- Node.js v18.2.0
- nodejs-polars v0.5.4

## Bug: Each time different arbitrary assignment of data to columns

Given the code

```js
const df = new DataFrame([
  ["a", new Date("2022-06-08T00:00:01"), 0],
  ["a", new Date("2022-06-08T00:00:06"), 2],
  ["a", new Date("2022-06-08T00:00:11"), 3],
  ["a", new Date("2022-06-08T00:00:16"), 6],
  ["a", new Date("2022-06-08T00:00:21"), 2],
  ["a", new Date("2022-06-08T00:00:26"), 4],
  ["b", new Date("2022-06-08T00:00:31"), 7],
  ["b", new Date("2022-06-08T00:00:36"), 9],
  ["b", new Date("2022-06-08T00:00:41"), 3],
  ["b", new Date("2022-06-08T00:00:46"), 4],
  ["b", new Date("2022-06-08T00:00:51"), 6],
  ["b", new Date("2022-06-08T00:00:56"), 7],
], { orient: "row", columns: ["group", "ts", "value"] })
```

When executed several times the data is assigned to columns each time in a different arbitrary order resulting in one of the below outcomes

### Incorrect behavior

```
shape: (12, 3)
┌───────┬─────┬─────────────────────┐
│ group ┆ ts  ┆ value               │
│ ---   ┆ --- ┆ ---                 │
│ f64   ┆ str ┆ datetime[ms]        │
╞═══════╪═════╪═════════════════════╡
│ 0.0   ┆ a   ┆ 2022-06-07 22:00:01 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 2.0   ┆ a   ┆ 2022-06-07 22:00:06 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 3.0   ┆ a   ┆ 2022-06-07 22:00:11 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 6.0   ┆ a   ┆ 2022-06-07 22:00:16 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ ...   ┆ ... ┆ ...                 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 3.0   ┆ b   ┆ 2022-06-07 22:00:41 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 4.0   ┆ b   ┆ 2022-06-07 22:00:46 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 6.0   ┆ b   ┆ 2022-06-07 22:00:51 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 7.0   ┆ b   ┆ 2022-06-07 22:00:56 │
└───────┴─────┴─────────────────────┘
```

### Incorrect behavior

```
shape: (12, 3)
┌───────┬─────┬─────────────────────┐
│ group ┆ ts  ┆ value               │
│ ---   ┆ --- ┆ ---                 │
│ str   ┆ f64 ┆ datetime[ms]        │
╞═══════╪═════╪═════════════════════╡
│ a     ┆ 0.0 ┆ 2022-06-07 22:00:01 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ a     ┆ 2.0 ┆ 2022-06-07 22:00:06 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ a     ┆ 3.0 ┆ 2022-06-07 22:00:11 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ a     ┆ 6.0 ┆ 2022-06-07 22:00:16 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ ...   ┆ ... ┆ ...                 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ b     ┆ 3.0 ┆ 2022-06-07 22:00:41 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ b     ┆ 4.0 ┆ 2022-06-07 22:00:46 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ b     ┆ 6.0 ┆ 2022-06-07 22:00:51 │
├╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ b     ┆ 7.0 ┆ 2022-06-07 22:00:56 │
└───────┴─────┴─────────────────────┘
```
### Correct behaviour

```
shape: (12, 3)
┌───────┬─────────────────────┬───────┐
│ group ┆ ts                  ┆ value │
│ ---   ┆ ---                 ┆ ---   │
│ str   ┆ datetime[ms]        ┆ f64   │
╞═══════╪═════════════════════╪═══════╡
│ a     ┆ 2022-06-07 22:00:01 ┆ 0.0   │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┤
│ a     ┆ 2022-06-07 22:00:06 ┆ 2.0   │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┤
│ a     ┆ 2022-06-07 22:00:11 ┆ 3.0   │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┤
│ a     ┆ 2022-06-07 22:00:16 ┆ 6.0   │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┤
│ ...   ┆ ...                 ┆ ...   │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┤
│ b     ┆ 2022-06-07 22:00:41 ┆ 3.0   │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┤
│ b     ┆ 2022-06-07 22:00:46 ┆ 4.0   │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┤
│ b     ┆ 2022-06-07 22:00:51 ┆ 6.0   │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┤
│ b     ┆ 2022-06-07 22:00:56 ┆ 7.0   │
└───────┴─────────────────────┴───────┘
```

**Comment by universalmind303** [comment]
This is a result of using hashmaps in the `infer_schema` function within `frame::row`. Hashmaps don't maintain order, I'll work on creating an indexmap equivalent that will maintain the column order. 

I should be able to include it in the next release.


in the meantime, you can map them using a `select` statement. 
```js
import pl from 'nodejs-polars'

const df = pl.DataFrame([
  ["a", new Date("2022-06-08T00:00:01"), 0],
  ["a", new Date("2022-06-08T00:00:06"), 2],
  ["a", new Date("2022-06-08T00:00:11"), 3],
  ["a", new Date("2022-06-08T00:00:16"), 6],
  ["a", new Date("2022-06-08T00:00:21"), 2],
  ["a", new Date("2022-06-08T00:00:26"), 4],
  ["b", new Date("2022-06-08T00:00:31"), 7],
  ["b", new Date("2022-06-08T00:00:36"), 9],
  ["b", new Date("2022-06-08T00:00:41"), 3],
  ["b", new Date("2022-06-08T00:00:46"), 4],
  ["b", new Date("2022-06-08T00:00:51"), 6],
  ["b", new Date("2022-06-08T00:00:56"), 7],
], {orient: "row"})
const columns = ["group", "ts", "value"];

console.log(df.select(columns.map((col, i) => pl.col(i.toString()).alias(col))))
```

**Comment by universalmind303** [comment]
closed via #3984
